modules/authentication/snip-enabling-user-authentication-with-github-common-first-steps.adoc

```diff
@@ -0,0 +1,128 @@
+:_mod-docs-content-type: SNIPPET
+.Procedure
+. Allow {product-short} to authenticate with GitHub, by creating a GitHub App.
++
+Use a GitHub App instead of an OAuth app to use fine-grained permissions, use short-lived tokens, scale with the number of installations by avoiding rate limits, and have a more transparent integration by avoiding to request user input.
+
+.. link:https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app[Register a GitHub App] with the following configuration:
+
+GitHub App name::
+Enter a unique name identifying your GitHub App, such as `authenticating-with-rhdh-__<GUID>__`.
+
+Homepage URL::
+Enter your {product-short} URL: `pass:c,a,q[{my-product-url}]`.
+
+Authorization callback URL::
+Enter your {product-short} authentication backend URL: `pass:c,a,q[{my-product-url}/api/auth/github/handler/frame]`.
+
+Webhook::
+Clear "Active", as this is not needed for authentication and catalog providers.
```

Comment:
"as this is not needed for authentication and catalog providers"

technically, this is fluff in the context of this procedure